CMakeLists.txt

```diff
@@ -0,0 +1,169 @@
+else()
+  message(WARNING "Not using Kokkos for tools configuration\n\tNote: using settings from installed is Kokkos not yet supported")
+endif()
+  # TODO: Configure based on Kokkos
+#       * check+test compile flags, especially ABI relevant
+#       * use installed version if Kokkos_ROOT or Kokkos_DIR is defined
+#       * check if Kokkos was already included in build with add_subdirectory()
+#       * error out otherwise: does it make sense to look into Kokkos
+#         source code without knowing it's *actual* build options ?
+
+#Turn MPI support off: (arbitrary or if Kokkos has MPI disabled)
+#CFLAGS += -DUSE_MPI=0
+
+# CXXFLAGS=-O3 -std=c++11 -g -shared -fPIC
+# -Wall -Wextra
+
+set(CMAKE_CXX_COMPILER ${Kokkos_CXX_COMPILER})
+set(CMAKE_CXX_STANDARD ${CMAKE_CXX_STANDARD_DEFAULT})
+
+# Libraries
+find_package(PAPI)
+if(PAPI_FOUND)
+  cmake_path(GET PAPI_INCLUDE_DIR PARENT_PATH PAPI_ROOT)
+endif()
+find_package(OpenMP)
+find_package(MPI)
+
+# Utilities
+add_subdirectory(common/kernel-filter)
+add_subdirectory(debugging/kernel-logger)
+
+# Profilers
+add_subdirectory(profiling/memory-events)
+add_subdirectory(profiling/memory-hwm)
+add_subdirectory(profiling/memory-hwm-mpi)
+add_subdirectory(profiling/memory-usage)
+add_subdirectory(profiling/chrome-tracing)
+add_subdirectory(profiling/simple-kernel-timer-json)
+add_subdirectory(profiling/simple-kernel-timer)
+add_subdirectory(profiling/space-time-stack)
+
+# External lib connectors
+if(PAPI_FOUND)
+  add_subdirectory(profiling/papi-connector)
+else()
+  message(WARNING "PAPI library not found - skipping papi-connector")
+endif()
+
+#add_subdirectory(profiling/systemtap-connector)
+add_subdirectory(profiling/variorum-connector)
+
+# Build external
+# TODO: tpls/apex
+
+# GPU profilers
+if(KOKKOS_ENABLE_CUDA)
+  #add_subdirectory(profiling/nvprof-connector)
+  #add_subdirectory(profiling/nvprof-focused-connector)
+endif()
+if(KOKKOS_ENABLE_HIP)
+  #add_subdirectory(profiling/roctx-connector)
+endif()
+
+if(DEFINED ENV{VTUNE_HOME})
+  set(VTune_ROOT $ENV{VTUNE_HOME})
+endif()
+if(VTune_ROOT)
+  #add_subdirectory(profiling/vtune-connector)
+  #add_subdirectory(profiling/vtune-focused-connector)
+else()
+  message(WARNING "Set VTUNE_HOME in environment or VTune_ROOT in build options to build VTune connectors")
+  set(VTune_ROOT "" CACHE STRING "Path to VTune Intel compiler")
+endif()
+
+# Note: Caliper docs https://software.llnl.gov/Caliper/build.html
+#       might miss some options, check the top CMakeLists.txt:
+#       https://github.com/LLNL/Caliper/blob/master/CMakeLists.txt
+macro(configure_caliper)
+  #set(BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS}) # Build shared (ON - default) or static library (OFF).
+  #set(BUILD_DOCS OFF)   # Build documentation.
+  #set(BUILD_TESTING OFF) # Build unit tests.
+  #set(WITH_FORTRAN OFF) # Build and install Fortran wrappers.
+  set(WITH_KOKKOS ON) # Enable Kokkos profiling support
```

Comment:
So I added support in Caliper to do:

```cmake
set(CALIPER_OPTION_PREFIX ON CACHE INTERNAL "Prefix caliper options with CALIPER_" FORCE)
```

and all the caliper options, e.g. `WITH_KOKKOS` appear in the cache as `CALIPER_WITH_KOKKOS`. This makes it very clear what are caliper options when it is a subproject.

Reply:
Great, thanks! Options prefixed.